packages/web/components/complex/asset-fieldset.tsx

```diff
@@ -0,0 +1,295 @@
+  <div className="flex w-full items-center justify-between pt-3">
+    {children}
+  </div>
+);
+
+const AssetFieldsetHeaderLabel = ({ children }: PropsWithChildren<unknown>) =>
+  children;
+
+const AssetFieldsetHeaderBalance = observer(
+  ({
+    onMax,
+    availableBalance,
+    className,
+    showAddFundsButton,
+    openAddFundsModal,
+    isMaxButtonDisabled,
+    isLoadingMaxButton,
+  }: {
+    onMax?: () => void;
+    availableBalance?: ReactNode;
+    className?: string;
+    showAddFundsButton?: boolean;
+    openAddFundsModal?: () => void;
+    isMaxButtonDisabled?: boolean;
+    isLoadingMaxButton?: boolean;
+  }) => {
+    const { t } = useTranslation();
+    const { accountStore } = useStore();
+
+    const wallet = accountStore.getWallet(accountStore.osmosisChainId);
+
+    return (
+      <div
+        className={classNames(
+          "flex items-center gap-2 transition-opacity",
+          className
+        )}
+      >
+        {wallet?.isWalletConnected ? (
+          showAddFundsButton ? (
+            <button
+              type="button"
+              onClick={openAddFundsModal}
+              className="body2 flex items-center justify-center rounded-5xl bg-wosmongton-700 py-1.5 px-3"
+            >
+              {t("limitOrders.addFunds")}
+            </button>
+          ) : (
+            <>
+              <span className="body2 text-osmoverse-300">
+                {availableBalance} {t("pool.available").toLowerCase()}
+              </span>
+              {onMax && (
+                <button
+                  type="button"
+                  className="flex items-center justify-center gap-1 rounded-5xl border border-osmoverse-700 py-1.5 px-3 disabled:pointer-events-none disabled:opacity-50"
+                  onClick={onMax}
+                  disabled={isMaxButtonDisabled}
+                >
+                  {isLoadingMaxButton && (
+                    <Spinner className="!h-2.5 !w-2.5 text-wosmongton-300" />
+                  )}
+                  <span className="body2 text-wosmongton-300">Max</span>
+                </button>
+              )}
+            </>
+          )
+        ) : null}
+      </div>
+    );
+  }
+);
+
+const calcFontSize = (numChars: number, isMobile: boolean): string => {
+  const sizeMapping: { [key: number]: string } = isMobile
+    ? {
+        9: "48px",
+        15: "38px",
+        24: "32px",
+        100: "16px",
+      }
+    : {
+        7: "48px",
+        12: "38px",
+        16: "28px",
+        33: "24px",
+        100: "16px",
+      };
+
+  for (const [key, value] of Object.entries(sizeMapping)) {
+    if (numChars <= Number(key)) {
+      return value;
+    }
+  }
+
+  return "1";
+};
+
+interface AssetFieldsetInputProps {
+  inputPrefix?: ReactNode;
+  onInputChange?: ChangeEventHandler<HTMLInputElement>;
+  inputValue?: string;
+  outputValue?: ReactNode;
+  page?: EventPage;
+}
+
+const AssetFieldsetInput = forwardRef<
```

Comment:
This might require a bit more changes. I've made a [ticket](https://linear.app/osmosis/issue/LIM-260/replace-assetfieldinput-with-scaledcurrencyinput) to track changing this post launch due to time restrictions.